### PR TITLE
support limit joiner's maxRedeemers

### PR DESCRIPTION
**Pull request by smallst**
limit the maxRedeemers. note that maxRedeemers unset/ set to 0 / set to prepaid.maxRedeemers means no limit, so they still see full number of prepaid.maxRedeemers

image
image

### CODE REVIEW

File: app/views/teachers/ShareLicensesStoreModule.coffee [hunk 1]
```diff
@@ -32,6 +32,13 @@ module.exports = ShareLicensesStoreModule =
       })
     revokeTeacher: (state, joinerID) ->
       state._prepaid.joiners = _.filter state._prepaid.joiners, (j) -> j._id != joinerID
+    updateTeacher: (state, {userID, maxRedeemers}) ->
+      state._prepaid.joiners = _.map state._prepaid.joiners, (j) ->
+        if j._id == userID
+          j.maxRedeemers = maxRedeemers
+          if maxRedeemers <= 0 || maxRedeemers >= state._prepaid.maxRedeemers
+            j.maxRedeemers = state._prepaid.maxRedeemers
```
[the remark this answers is not in the document]
Comment (smallst) [reply app/views/teachers/ShareLicensesStoreModule.coffee:40]: oh you catch it! yeah i forget we could update prepaid.maxRedeemers! thanks!

[the remark this answers is not in the document]
Comment (smallst) [reply app/views/teachers/ShareLicensesStoreModule.coffee:40]: image
check here

[the remark this answers is not in the document]
Comment (mrfinch) [reply app/views/teachers/ShareLicensesStoreModule.coffee:40]: I don't think it's intuitive that someone will set it to -ve or a large number. Maybe add a button for it or just remove it. They can obviously delete sharing and re-add it instead

[the remark this answers is not in the document]
Comment (smallst) [reply app/views/teachers/ShareLicensesStoreModule.coffee:40]: yeah they can delete sharing/re-add . we can also limit the number input as min=1, max=MaxRedeemers. i just think add another button makes UI crash (no left space now). actually 0 for unlimited or MaxRedeemers for unlimited is reasonable. while yeah, negative or really really large is stupid and maybe no one would set that. :joy: but the logic is still safe in these situations. 😂

**Review by mrfinch — APPROVED** [review]

**Pull request merged**